fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -326,8 +324,8 @@ private FHIRRestOperationResponse doPatchOrUpdate(String type, String id, FHIRPa
                     ior.setPrevResource(null);
 
                     // if no id provided, then generate an id for the input resource
-                    if (newResource.getId() == null || newResource.getId() == null) {
-                        id = UUID.randomUUID().toString();
+                    if (newResource.getId() == null) {
```

Comment:
strange logic there, generally when I see that I wonder why there were two, is newResource ever null? was the first check in case newResource was null?

Reply:
newResource should not be null at this point - it's checked during bundle validation (or resource validation if request is not a bundle request)